### PR TITLE
Linux installer

### LINKED ISSUE #2: Missing Linux installer
Hi @bryanmacfarlane !

There is no Linux installer, and what I understand from (this issue)[https://github.com/Microsoft/vso-agent/issues/105] in the vso-agent-installer package is that you are planning to use docker to support Linux.

I have successfully run the vso agent on Linux as a service with `systemd` and without docker, if I create the Linux installer based on systemd and I make a PR to this package, will you merge it if is fine?

If you aim is to not add that kind of support and you will not merge it, I will not spend my time doing it :P

### PR DESCRIPTION
**Pull request by artberri**
- `lib/installers/Linux.js` added
- README.md updated (change init.d with systemd)
- Added bash header to svc to run properly on linux.
- Fixes #2

### CODE REVIEW

File: lib/installers/Linux.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+  config.name = name;
+  config.runAsUser = service.user;
+  config.logFolder = 'syslog';
+  config.definition = service;
+
+  unit += '[Service]' + "\n";
+  unit += 'ExecStart=' + service.command + "\n";
+  unit += 'Restart=always' + "\n";
+  unit += 'StandardOutput=syslog' + "\n";
```
Comment (bryanmacfarlane) [inline lib/installers/Linux.js]: Is journald a better choice?  I believe if not specified, then journald is chosen and that was created by the folks that created systemd.  Is there a reason to go with syslog?

File: lib/installers/Linux.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  unit += 'StandardError=syslog' + "\n";
+  unit += 'SyslogIdentifier=' + service.logIdentifier + "\n";
+  unit += 'User=' + service.user + "\n";
+  unit += 'Group=' + service.user + "\n";
+  unit += 'Environment=NODE_ENV=production' + "\n";
+  unit += 'Environment=PATH=' + service.path + "\n";
+  unit += "\n";
+  unit += '[Install]' + "\n";
+  unit += 'WantedBy=multi-user.target' + "\n";
+
```
Comment (bryanmacfarlane) [inline lib/installers/Linux.js:156]: Should we depend on network online?  I guess that could be an option
http://unix.stackexchange.com/questions/126009/cause-a-script-to-execute-after-networking-has-started

**Comment by artberri** [comment]
- I will use `journal`, you are right
- I will double check  about depending on network online

But I will also make some other changes, for example I will remove `NODE_ENV=production`, it causes node to not install devDependencies. It's a build server/agent, it is not production.

I'm checking it in a project that we are moving from Jenkins to VSO, once I have it working properly I'll came back with the fixes.

Thanks!

**Comment by artberri** [comment]
Hi again @bryanmacfarlane !

I think that now it's done. We have been running the vso-agent in Linux servers during a while (with systemd) successfuly, and we are now deploying to production with it also, (new VSTS Releases feature). Our migration from Jenkins was fine.

**Comment by bryanmacfarlane** [comment]
Cool.  I'll merge

Note that we have a new agent replacing the node.  
https://github.com/microsoft/vsts-agent

**Comment by artberri** [comment]
Cool!

Yeah! I saw it the some days ago. We will wait until it gets out from beta to use it, anyway I'll start doing some tests.

It will replace the node.js one, but... Is it intended to replace also the Windows one?

**Comment by bryanmacfarlane** [comment]
Yes.  It will be the one agent.

**Pull request merged**